src/Lecture2.hs

```diff
@@ -160,7 +175,62 @@ data Knight = Knight
     , knightEndurance :: Int
     }
 
-dragonFight = error "TODO"
+data DragonType =
+  Red | Green | Black
+  deriving (Eq)
+
+data Dragon = Dragon
+    {dragonHealth :: Int
+    ,dragonAttack :: Int
+    ,dragonType   :: DragonType
+    ,dragonChest  :: Chest
```

Comment:
That's a great idea! And almost the perfect solution 🔝 

There're two problems with this particular data type definition:

1. It's possible to have a treasure inside green dragon:

```haskell
myDragon = Dragon
    { dragonHealth = 150
    , dragonAttack = 70
    , dragonType = Green
    , dragonChest = Chest 87 (Just "Great Sword +2")
    }
```

2. It's possible to not have a treasure inside non-green dragons:

```haskell
myDragon = Dragon
    { dragonHealth = 150
    , dragonAttack = 70
    , dragonType = Red
    , dragonChest = Chest 87 Nothing
    }
```